nvtabular/column_selector.py

```diff
@@ -13,16 +13,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-from dataclasses import dataclass, field
 from typing import List
 
 import nvtabular
 
 
-@dataclass
 class ColumnSelector:
-    _names: List[str]
-    subgroups: List["ColumnSelector"] = field(default_factory=list)
+    def __init__(self, names: List[str], subgroups: List["ColumnSelector"] = None):
+        self._names = names
+        self.subgroups = subgroups if subgroups else []
+        self.__post_init__()
```

Comment:
I think we can inline the `__post_init__` method here in the constructor. `dataclass` calls that hook method, but it probably doesn't make sense as a separate method any more.